### PR TITLE
Timeline item menu

### PR DESCRIPTION
**Pull request by stefanceriu**
This PR implements the new timeline item "action" menu which, instead of a context menu, is a bottom sheet on the iPhone and a popover on the iPad, similar to the attachments picker

The new emoji section will come in a later PR

https://github.com/vector-im/element-x-ios/assets/637564/ea4f9614-8643-41e5-9d5c-d457aeefab98

Screenshot 2023-05-31 at 17 59 35

### CODE REVIEW

**Comment by callumu** [comment]
Also, could you make sure there is an 8 pixel space from the avatar and message preview contents?

**Comment by callumu** [comment]
> > Design comments
> 
> Yes to everything 👍
> 
> > What colour is the separator currently?
> 
> No idea, it's the system default, choice yours

Okay cool. Could you set it as the Compound Light/color/gray/400?

**Comment by stefanceriu** [comment]
> Is there a way that when going full screen, the bottom sheet doesn't turn into a modal but just increases in height?

That one I don't think we can do, that's just how the system component works

**Comment by callumu** [comment]
> > Is there a way that when going full screen, the bottom sheet doesn't turn into a modal but just increases in height?
> 
> That one I don't think we can do, that's just how the system component works

Okay no worries then

**Comment by stefanceriu** [comment]
I think this is closer to what we want: 

Screenshot 2023-06-06 at 12 42 34

**Review by alfogrillo — APPROVED** [review]
Nice change! 👍🏻
Just left some comments for minor improvements.

File: ElementX/Sources/Screens/RoomScreen/View/Style/TimelineItemBubbledStylerView.swift [hunk 1]
```diff
@@ -98,12 +100,23 @@ struct TimelineItemBubbledStylerView<Content: View>: View {
...
+                context.send(viewAction: .itemTapped(id: timelineItem.id))
+            }
+            .onTapGesture(count: 2) {
+                context.send(viewAction: .displayEmojiPicker(itemID: timelineItem.id))
+            }
+            // We need a tap gesture before this long one so that it doesn't
+            // steal away the gestures from the scroll view
+            .onLongPressGesture(minimumDuration: 0.25) {
+                showItemActionMenu = true
+                UIImpactFeedbackGenerator(style: .heavy).impactOccurred()
```
[the remark this answers is not in the document]
Comment (stefanceriu) [reply ElementX/Sources/Screens/RoomScreen/View/Style/TimelineItemBubbledStylerView.swift:118]: Very good idea 👍

**Comment by stefanceriu** [comment]
Unfortunately I am forced to move away from timeline item bound SwiftUI popovers because they randomly get closed when the timeline updates. I can't find anything that would make them stay put. 
The benefit they were bringing was that the bottom sheet would automatically be transformed into a popover for iPads. With this change the iPhone layout stays the same but the iPad one becomes this:

Screenshot 2023-06-06 at 18 44 56

We will need to decide if we want to build a custom bottom sheet that works the same on iPhone and iPad to work around this..

**Comment by callumu** [comment]
> Screenshot 2023-06-06 at 17 28 33
> 
> The background for the already selected one is a placeholder, let's talk about the design tomorrow Same for the "more" icon, that's not a SF symbol so should we make it available through compound? Don't think that's ready yet

I don't think it's ready in Compound yet, but can we use it even if it's not?

Yep let's talk about this more today and the iPad behaviour also

**Comment by stefanceriu** [comment]
Today we agreed that the iPad version is good for now but we needed to change the reactions a bit. Here's how it looks lie following my last commit

Screenshot 2023-06-07 at 17 48 44

**Comment by callumu** [comment]
> Today we agreed that the iPad version is good for now but we needed to change the reactions a bit. Here's how it looks lie following my last commit
> 
> Screenshot 2023-06-07 at 17 48 44

1. Could you reduce the spacing between the emojis and the dividers to 16px?
2. Are you able to increase the size of the selected state?

Aside from these two it looks good

**Comment by stefanceriu** [comment]
> Aside from these two

Screenshot 2023-06-07 at 18 06 28

Sure thing, I think this is what we're looking for ^, right?

**Comment by stefanceriu** [comment]
Talked about it offline and got the design just right.

Otherwise, Github somehow managed to break and it's no longer recognising updates to this PR, keeps asking to resolve conflicts although there are none anymore. Closing this PR and creating a new one.

**Pull request closed without merging**